packages/sdk/src/lib/dataProtectorSharing/getProtectedDataByCollection.ts

```diff
@@ -0,0 +1,115 @@
+import { gql } from 'graphql-request';
+import { toHex } from '../../utils/data.js';
+import { WorkflowError } from '../../utils/errors.js';
+import {
+  numberBetweenSchema,
+  positiveNumberSchema,
+  throwIfMissing,
+} from '../../utils/validators.js';
+import { ProtectedDatasGraphQLResponse } from '../types/graphQLTypes.js';
+import {
+  GetProtectedDataByCollectionParams,
+  ProtectedDataInCollection,
+  SubgraphConsumer,
+} from '../types/index.js';
+
+export const getProtectedDataByCollection = async ({
+  graphQLClient = throwIfMissing(),
+  collectionTokenId,
+  creationTimestampGte,
+  page = 0,
+  pageSize = 1000,
+}: SubgraphConsumer & GetProtectedDataByCollectionParams): Promise<
+  ProtectedDataInCollection[]
+> => {
+  const vCollectionTokenId = positiveNumberSchema()
+    .label('collectionTokenId')
+    .validateSync(collectionTokenId);
+  const vCreationTimestampGte = positiveNumberSchema()
+    .label('creationTimestampGte')
+    .validateSync(creationTimestampGte);
+
+  const vPage = positiveNumberSchema().label('page').validateSync(page);
+  const vPageSize = numberBetweenSchema(10, 1000)
+    .label('pageSize')
+    .validateSync(pageSize);
+
+  try {
+    const start = vPage * vPageSize;
+    const range = vPageSize;
+    const collectionTokenIdHex = toHex(vCollectionTokenId);
+
+    const SchemaFilteredProtectedData = gql`
+    query (
+      $start: Int!
+      $range: Int!
+    ) {
+      protectedDatas(
+        where: {
```

Comment:
I don't think. Because for me the user story is : 
- As a user i want to see my protectedData and people protectedData. 

This function should also enable builder to get it's protectedData inside its collection in order to call : setProtectedDataForSale, setProtectedDataToRenting, ...

May be a missing function could be added to return the statu of the protectedData (renting, sale, ....)

Reply:
Fair enough, indeed a collection's owner obviously wants to get all its protected data, whatever the distribution params they have.

I was thinking of the homepage where it's not so relevant to display a content card and when you click on it there is nothing you can do to visualise it. (not for rent, not for sale, not included in a subscription...)

Maybe we can include these 3 fields in the subgraph query:
 - isRentable
 - isIncludedInSubscription
 - isForSale

This way the caller can filter out as they wish.

Reply:
yeah good idea
fix Commit: ee52b00f87fb061fbc41c5a99af45b47eaeea196